crates/matrix-sdk/src/widget/matrix.rs

```diff
@@ -534,9 +535,13 @@ impl StateUpdateReceiver {
         loop {
             match self.room_updates.recv().await? {
                 RoomUpdate::Joined { room, updates } => {
-                    if !updates.state.is_empty() {
-                        return Ok(updates
-                            .state
+                    let state_events = match updates.state {
+                        State::Before(events) => events,
+                        State::After(events) => events,
+                    };
```

Comment:
We are losing the information here about whether these state events cover the whole update or not. Should the widget API take the result of `collect`, so it can always just update state based on what we pass it?

Reply:
I really don't know, I guess that it depends on what the widget is expecting. But the information was already lost before between sync v2 and SSS.